cmake/scripts/nbl/gitInfo.cmake

```diff
@@ -0,0 +1,18 @@
+# A configure-time and build-time script, that produces source with current git commit info
```

Comment:
> (...) build-time script

actually we should not proxy pass the script to be executed at build time nor create a build target for that purpose - it can be all done nicely at configure time without introducing special target to the build system

we could just `configure_file` and that will do the job, I would remove the script + `CheckGit` target and move the script's content directly where you include it

by the way if it complains about `"${OUTPUT_FILE}"` being not `GENERATED` when you create the library with it as part of your sources then you can simply set [this property](https://cmake.org/cmake/help/latest/prop_sf/GENERATED.html#prop_sf:GENERATED) for the output

Reply:
With build-time check we make sure that: for any "bug report" (from someone who builds Nabla on their own) will have correct git-hash.

In case of configure-time only: checking out different commit/branch without re-configuring CMake, we will keep the old hash.  For releases (i.e. `Nabla.dll`) config-time only is good enough.

Reply:
> checking out different commit/branch without re-configuring CMake, may keep the old hash

okay fair argument, but if we want to make it well then integrate [this](https://github.com/andrew-hardin/cmake-git-version-tracking/tree/master) repo as 3rdparty instead writing our own git check actually (it does exactly what you do now but with a few improvements regarding regenerating the .cpp if the git hash has actually changed + much more metadata there is already written we just plug it & it works -> you can read [this](https://jonathanhamberg.com/post/cmake-embedding-git-hash/) article for details why it's better and what they do extra)

Reply:
okay I forked the [repo](https://github.com/Devsh-Graphics-Programming/cmake-git-version-tracking) - use it & add short version of commit sha, the only thing to do then is to link the target to Nabla afterwards

Reply:
> with a few improvements regarding regenerating the .cpp if the git hash has actually changed

Actually we don't need their solution for this issue. I used CMake 3.10+ feature for this issue (our minimal: 3.29). This gives much cleaner script (see [gitInfo.cmake](https://github.com/Devsh-Graphics-Programming/Nabla/blob/16f83bf555146637528e11c2c2d9d13a3afb3c95/cmake/scripts/nbl/gitInfo.cmake)).

Reply:
explained in discord, we stay with the fork + introduce a small modification to it

Reply:
- add short sha
- modify the template cpp & header to keep our meta data in nice structure
- we need to use NBL_API2 for the output header's structure, so we just have to include our define.h
- link to Nabla, profit

Reply:
sorry, the `NBL_API2` is in [BuildConfigOptions.h](https://github.com/Devsh-Graphics-Programming/Nabla/blob/a10ab3d0ce1626a1d5ea8ed7d2e8a0aca4665462/include/nbl/config/BuildConfigOptions.h.in#L91) but it's okay, lets give the meta library target proper [include search directory](https://cmake.org/cmake/help/latest/command/target_include_directories.html) and include the config

Reply:
as you are on it you can fix this [`// DDL exports`](https://github.com/Devsh-Graphics-Programming/Nabla/blob/a10ab3d0ce1626a1d5ea8ed7d2e8a0aca4665462/include/nbl/config/BuildConfigOptions.h.in#L90C1-L90C15) typo xD